src/config.rs

```diff
@@ -67,20 +74,43 @@ impl fmt::Display for ToolSpec {
     }
 }
 
+#[derive(Debug, Serialize)]
+pub struct ConfigFileTools(HashMap<String, ToolSpec>);
+
+impl ConfigFileTools {
+    pub fn new() -> ConfigFileTools {
+        Self(HashMap::new())
+    }
+}
+
+impl Deref for ConfigFileTools {
+    type Target = HashMap<String, ToolSpec>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for ConfigFileTools {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
 #[derive(Debug, Serialize, Deserialize)]
 pub struct ConfigFile {
-    pub tools: HashMap<String, ToolSpec>,
+    pub tools: ConfigFileTools,
```

Comment:
Since some external files (like `src/main.rs`) want to perform methods like `tools.get`, I think it would probably be best to just do
```rs
pub fn tools(&self) -> &HashMap<String, ToolSpec> {
    &self.tools.0
}
```
which would leave us to have to create a `tools_mut` as well (as `ConfigFile::fill_from` needs it), or we could leave it the current way it is (implement Deref and DerefMut).